src/store.ts

```diff
@@ -413,32 +409,115 @@ export default function Store(
                 this.SDKConfig.onCreateBatch = undefined;
             }
         }
+    }
+}
 
-        if (!config.hasOwnProperty('flags')) {
-            this.SDKConfig.flags = {};
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            )
-        ) {
-            this.SDKConfig.flags[
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            ] = Constants.DefaultConfig.uploadInterval;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.ReportBatching
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.ReportBatching] = false;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.OfflineStorage
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.OfflineStorage] = 0;
+export interface IFeatureFlags {
+    reportBatching?: string;
+    eventBatchingIntervalMillis?: string;
+    offlineStorage?: string;
+    directURLRouting?: boolean;
+}
+
+export function processFlags(
+    config: SDKInitConfig,
+    SDKConfig: SDKConfig
+): IFeatureFlags {
+    const flags: IFeatureFlags = {};
+    const {
+        ReportBatching,
+        EventBatchingIntervalMillis,
+        OfflineStorage,
+        DirectUrlRouting,
+    } = Constants.FeatureFlags;
+
+    if (!config.flags) {
+        return {};
+    }
+
+    // Passed in config flags take priority over defaults
+    flags[ReportBatching] = config.flags[ReportBatching] || false;
+    flags[EventBatchingIntervalMillis] =
+        config.flags[EventBatchingIntervalMillis] ||
+        Constants.DefaultConfig.uploadInterval;
+    flags[OfflineStorage] = config.flags[OfflineStorage] || '0';
+    flags[DirectUrlRouting] = config.flags[DirectUrlRouting] === 'True';
+
+    return flags;
+}
+
+export function processEndpoints(
+    config: SDKInitConfig,
+    flags: IFeatureFlags,
+    apiKey: string
+): Dictionary {
+    // an API key is not present in a webview only mode. In this case, no endpoints are needed
+    if (!apiKey) {
+        return {};
+    }
+
+    // Set default endpoints
+    let endpoints: Dictionary = JSON.parse(
```

Comment:
You're never reassigning, so this can be `const`.

```suggestion
    const endpoints: Dictionary = JSON.parse(
```